src/server.ts

```diff
@@ -92,8 +98,56 @@ export const start = (options: StartOptions) => {
 
   apolloServer.applyMiddleware({ app, path: '/' });
 
-  const httpServer = http.createServer(app);
-  httpServer.listen(settings.port)
-  
+  const httpServer = http.createServer(app).listen(settings.port);
+
+  startWebSocketServer(httpServer)
+
   console.log(`Server is running on http://localhost:${settings.port}`);
-};
\ No newline at end of file
+};
+
+const startWebSocketServer = (httpServer: net.Server) => {
+  const wss = new Server({
+    noServer: true,
+    perMessageDeflate: false
+  });
+
+  // expecting path /langServer/<lang_name>, e.g /langServer/java
+  httpServer.on('upgrade', (request: http.IncomingMessage, socket: net.Socket, head: Buffer) => {
+    const pathname = request.url ? url.parse(request.url).pathname : undefined;
+    if (pathname.startsWith('/langServer')) {
```

Comment:
negate to avoid indentation  
also, should probably log and return error and disconnect when path isn't right?